sklearn/discriminant_analysis.py

```diff
@@ -378,7 +378,9 @@ def _solve_lsqr(self, X, y, shrinkage, covariance_estimator):
         self.covariance_ = _class_cov(
             X, y, self.priors_, shrinkage, covariance_estimator
         )
-        self.coef_ = linalg.lstsq(self.covariance_, self.means_.T)[0].T
+        self.coef_ = linalg.lstsq(self.covariance_, self.means_.T, check_finite=False)[
```

Comment:
So I checked the line above, and I don't think that we can get a non-finite covariance since `bias=1` in the `np.cov` would avoid a division by zero that is the only way to get such of non-finite covariance.